yarn-project/noir-contracts/src/contracts/token_contract/src/types.nr

```diff
@@ -0,0 +1,176 @@
+use dep::std::hash::pedersen;
+use dep::std::hash::pedersen_with_separator;
+use dep::aztec::note::{
+    note_header::NoteHeader,
+    note_interface::NoteInterface,
+    utils::compute_siloed_note_hash,
+};
+use dep::aztec::constants_gen::GENERATOR_INDEX__L1_TO_L2_MESSAGE_SECRET;
+
+global TRANSPARENT_NOTE_LEN: Field = 2;
+
+
+
+struct AztecAddress {
+  address: Field
+}
+
+impl AztecAddress {
+  fn new(address: Field) -> Self {
+    Self {
+      address
+    }
+  }
+
+  fn serialize(self: Self) -> [Field; 1] {
+    [self.address]
+  }
+
+  fn deserialize(fields: [Field; 1]) -> Self {
+    Self {
+      address: fields[0]
+    }
+  }
+}
+
+struct EthereumAddress {
+  address: Field
+}
+
+impl EthereumAddress {
+  fn new(address: Field) -> Self {
+    Self {
+      address
+    }
+  }
```

Comment:
`AztecAddress` and `EthereumAddress` should be added to `aztec-nr/` instead, since they're widely applicable types.